examples/qwen_3_vl/qwen_3_vl.zig

```diff
@@ -0,0 +1,1211 @@
+    pub fn forward(
+        self: Qwen3VL,
+        image_hwc: Tensor,
+        input_ids: Tensor,
+        image_dim: Tensor,
+        token_index: Tensor,
+        prompt_shape: Tensor,
+        kv_cache: KvCache,
+        h_resized: u32,
+        w_resized: u32,
+        rng: Tensor.Rng,
+    ) struct { Tensor, KvCache, Tensor, Tensor.Rng } {
+        const pixel_value, const image_grid_thw = self.processImage(image_hwc, image_dim, h_resized, w_resized);
+        const next_token, const updated_cache, const mrope_position_deltas, const new_rng = zml.call(self.qwen, .forward, .{ input_ids, pixel_value, token_index, image_grid_thw, kv_cache, prompt_shape, rng });
+
+        return .{ next_token, updated_cache, mrope_position_deltas, new_rng };
+    }
+
+    pub fn processImage(
+        self: Qwen3VL,
+        image_hwc: Tensor,
+        image_size: Tensor,
+        h_resized: u32,
+        w_resized: u32,
+    ) struct { Tensor, [3]u32 } {
+
+        // Resize the image and transpose (channels, height, width)
+        var image_chw = ResizeBicubic(image_hwc, .{ .h = h_resized, .w = w_resized }, .{ .original_len = image_size }).transpose(.{ 2, 0, 1 });
+        image_chw = image_chw.convert(.f32);
+
+        // Rescale and normalize the image
+        const rescale_factor: f32 = 1.0 / 255.0;
+        const image_mean: f32 = 0.5;
+        const image_std: f32 = 0.5;
+        image_chw = image_chw.scale(rescale_factor); // pixel / 255.0
+        var image_chw_rescaled_normalized = image_chw.sub(Tensor.scalar(image_mean, .f32)).div(Tensor.scalar(image_std, .f32));
+
+        // Introduce the temporal dimension (1)
+        image_chw_rescaled_normalized = image_chw_rescaled_normalized.reshape(.{ .c = 3, .temporal_patch_size = 1, .h = h_resized, .w = w_resized });
+        const temporal_patch_size = self.qwen.config.vision_config.temporal_patch_size;
+
+        // Repeat the image 2 times in the temporal dimension
+        image_chw_rescaled_normalized = image_chw_rescaled_normalized.repeat1d(1, 2);
+        const patch_size = self.qwen.config.vision_config.patch_size;
+
+        //Hardcoded because we only have 1 temporal patch (image)
+        const grid_t = 1;
+        // Compute the number of grid cells based on the patch size (size of the patch embedding)
+        const grid_h: u32 = @intCast(@as(u32, @divExact(h_resized, patch_size)));
+        const grid_w: u32 = @intCast(@as(u32, @divExact(w_resized, patch_size)));
+        const grid_thw = [3]u32{ grid_t, grid_h, grid_w };
+
+        const merge_size = self.qwen.config.vision_config.spatial_merge_size;
+
+        // Reshape the image to the correct shape for the vision transformer
+        image_chw_rescaled_normalized = image_chw_rescaled_normalized.reshape(.{ .temporal_patch_size = temporal_patch_size, .c = 3, .h_div = @divExact(grid_h, merge_size), .m1 = merge_size, .patch1 = patch_size, .w_div = @divExact(grid_w, merge_size), .m2 = merge_size, .patch2 = patch_size });
```

Comment:
in general `reshape( divisions )` is better served by a `.splitAxis`: no room for errors